File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let map = {};

    for(let i = 0; i < nums.length; i++){
        map[nums[i]] = i;
    }

    for(let i = 0; i < nums.length; i++){
        let val = target - nums[i];
        if(map[val] != i && val in map){
            return[i, map[val]];
        }
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
		
		Map<Integer, Integer> map = new HashMap<Integer, Integer>();

		for(int i = 0; i < nums.length; i++) {
			int searchVal = target - nums[i];
			
			if(map.containsKey(searchVal)) {
				return new int[] {map.get(searchVal), i};
			}
			
			map.put(nums[i], i);
		}
		
		System.out.println(map);

		
		return new int[] {69, 96};
	}

}